ui/core-react/src/test/elementseparator/ElementSeparator.test.tsx

```diff
@@ -116,108 +152,114 @@ describe("ElementSeparator", () => {
+      it("stops calling onRatioChanged when dragging stops", async () => {
+        render(
           <ElementSeparator
             orientation={Orientation.Horizontal}
             movableArea={100}
             ratio={0.5}
             onRatioChanged={onRatioChanged}
           />);
 
-        elementSeparator.simulate("pointerdown", { clientX: 50 });
-        moveElement({ clientX: 70 });
+        await theUserTo.pointer([{keys: "[MouseLeft>]", target: getButton(), coords: {x: 50}},
+          {coords: {x: 70}},
+        ]);
 
         expect(onRatioChanged.callCount).to.be.equal(1);
 
-        document.dispatchEvent(new MouseEvent("pointerup"));
-        moveElement({ clientX: 90 });
+        await theUserTo.pointer([{keys: "[/MouseLeft]"},
+          {coords: {x: 90}},
+        ]);
 
         expect(onRatioChanged.callCount, "Called when dragging stopped").to.be.equal(1);
       });
 
-      it("does not call onRatioChanged when dragging without movableArea set", () => {
-        const elementSeparator = mount(
+      it("does not call onRatioChanged when dragging without movableArea set", async () => {
+        render(
           <ElementSeparator
             orientation={Orientation.Horizontal}
             ratio={0.5}
             onRatioChanged={onRatioChanged}
           />);
 
-        elementSeparator.simulate("pointerdown", { clientX: 50 });
-        moveElement({ clientX: 70 });
+        await theUserTo.pointer([{keys: "[MouseLeft>]", target: getButton(), coords: {x: 50}},
+          {coords: {x: 70}},
+        ]);
 
         expect(onRatioChanged.callCount).to.be.equal(0);
       });
 
-      it("stops calling onRatioChanged when pointerdown event happens while still dragging", () => {
-        const elementSeparator = mount(
+      it("stops calling onRatioChanged when pointerdown event happens while still dragging", async () => {
+        render(
           <ElementSeparator
             orientation={Orientation.Horizontal}
             movableArea={100}
             ratio={0.5}
             onRatioChanged={onRatioChanged}
           />);
 
-        elementSeparator.simulate("pointerdown", { clientX: 50 });
-        moveElement({ clientX: 70 });
+        await theUserTo.pointer([{keys: "[MouseLeft>]", target: getButton(), coords: {x: 50}},
+          {coords: {x: 70}},
+        ]);
 
         expect(onRatioChanged.callCount).to.be.equal(1);
 
-        elementSeparator.simulate("pointerdown", { clientY: 70 });
-        moveElement({ clientX: 90 });
+        // This is not an interaction that user-events allow or physically make any sense,
+        // you cant press your mouse button while it's pressed, but the code have a case for it.
+        // I'm leaving it here in case this is a know edge case that happens in a way I cant imagine now.
+        fireEvent.pointerDown(getButton());
+
+        await theUserTo.pointer([{target: getButton(), coords: {x: 70}},
+          {coords: {x: 90}}]);
```

Comment:
This test is one of the case I mentioned, where it is not "possible" as far as I can think of, to click while clicking, but fireEvent allow that since we are firing random event. The code is actually checking this situation, so I kept the test there.